lib/spec.js

```diff
@@ -20,8 +20,15 @@ function getReleaseNumber(release) {
 
 function getVersionNumber({ spec, version }) {
   const replaceHyphens = getValueFromSpec(spec, 'replaceHyphens', false);
+  let newVersion;
 
-  return replaceHyphens ? version.replace(/-/g, '~') : version;
+  if (replaceHyphens === true) {
+    newVersion = version.replace(/-/g, '~');
+  } else if (typeof replaceHyphens === 'string') {
```

Comment:
Good point - yes it will. My initial thinking was that the user can do what they want; it's their package and build spec, if they want to produce nonsensical version numbers they can go for it. But I suppose it leaves the behaviour of speculate itself a bit unpredictable. I'll come up with a list and add it in.

Reply:
Ok, have added filtering to allow tilde, underscore and plus [~_+]. And a console warning to let you know if you use another string that it hasn't been replaced. Let me know what you think :)